### ISSUE TITLE
devtools Object Snapshot are not display in tracing

### ISSUE THREAD

**Issue by fmeawad** [catapult-project/catapult#2223]
Record a trace with "disabled-by-default-devtools.timeline.picture" category enabled.
Click on one of the green dots that represents an Object Snapshot
Ouptut: Details panel is empty
Expected output: Details panel showing the Object Snapshot

JS Error: 
```javascript
Uncaught TypeError: Cannot read property 'forEach' of undefined  @ display_item_debugger.html:208
set displayItemList @ display_item_debugger.html:208
updateContents @ display_item_list_view.html:38
set objectSnapshot @ object_snapshot_view.html:42
set modelEvent @ object_snapshot_view.html:29
set selection @ single_object_snapshot_sub_view.html:83
onSelectionChanged_ @ analysis_view.html:179
dispatchEvent @ event_target.html:111
dispatchChangeEvent_ @ brushing_state_controller.html:41
set currentBrushingState @ brushing_state_controller.html:163
changeSelectionFromTimeline @ brushing_state_controller.html:225
onEndSelection_ @ timeline_track_view.html:1019
onMouseUp_ @ mouse_mode_selector.html:378
cleanupAndDispatchToMouseUp @ mouse_tracker.html:101
```

**Comment by chiniforooshan** [comment]
Could it be because of [this](https://codereview.chromium.org/1755413003) bug? If you send me the trace I can check if patching that CL fixes the issue.

**Comment by fmeawad** [comment]
I do not think so, the C++ side should populate the "items" list [here](https://code.google.com/p/chromium/codesearch#chromium/src/cc/playback/display_item_list.cc&l=247). But I am suspecting that @primiano changes in this [CL](https://codereview.chromium.org/1354493003/) may have caused it not to be written properly.

You can find the trace I am testing with shared [in drive](https://drive.google.com/a/google.com/file/d/0B3PSJY916EOjQTJZWTB1Sm02Rjg/view?usp=sharing).

**Comment by primiano** [comment]
Hmm that was 6 months ago.
@petrcermak any ideas?

**Comment by petrcermak** [comment]
I don't think it's @primiano's change (there doesn't seem to be dot expansion involved here).

From what I can see, the <tt>items</tt> array might not be added to the dictionary if [<tt>include_items</tt> is false](https://code.google.com/p/chromium/codesearch#chromium/src/cc/playback/display_item_list.cc&l=246), which is the case when [the "cc.debug.display_items" tracing category is not enabled](https://code.google.com/p/chromium/codesearch#chromium/src/cc/playback/display_item_list.cc&l=43). 

Perhaps the <tt>include_items</tt> flag should also be set to true when "disabled-by-default-devtools.timeline.picture" is enabled, but I think it would be better to just make Trace Viewer handle missing <tt>items</tt> graciously. Is there any relationship between "disabled-by-default-devtools.timeline.picture" and "cc.debug.display_items"?

**Comment by fmeawad** [comment]
I confirm that you need both categories to get that displayed properly, it is counter-intuitive.
We can:
1. Require both categories to generate the list in the first place
2. Update the UI to display the Pictures only without the list if the list is not available (+ a warning)
3. Update the UI to indicate that both categories should be enabled to get the correct display list.

Regardless, if you are trying to reproduce this:
- make sure to have `--enable-skia-benchmarking` enabled
- It is very hard to reproduce because once both categories are enabled, the buffer overflows (or the json is too big to parse), so you either open an empty page or record a trace of less than .5 seconds.

**Comment by petrcermak** [comment]
1. Not sure what the precedent is here, but I would personally avoid introducing cross-dependencies between categories if possible.
2. This seems like the __best option__ to me: Make the UI more robust and tell the user what they need to do to get the list.
3. I don't see any benefit of this option over 2.
